### PR TITLE
Decompile Battle Transition

### PR DESCRIPTION
**Pull request by DizzyEggg**
First 15 transitions are basically the same, the champion one is different(Steven/Wallace) and there's a lot of new ones (17-41).

### CODE REVIEW

File: src/battle_transition.c [hunk 1]
```diff
@@ -0,0 +1,4346 @@
...
+#include "decompress.h"
+#include "bg.h"
+#include "blend_palette.h"
+
+struct TransitionData
+{
+    vs8 VBlank_DMA;
+    u16 WININ;
+    u16 WINOUT;
+    u16 field_6;
```
Comment (Diegoisawesome) [inline src/battle_transition.c]: Wouldn't this be WIN0H?

Comment (DizzyEggg) [reply src/battle_transition.c]: seems like it

File: src/battle_transition.c [hunk 2]
```diff
@@ -0,0 +1,4346 @@
...
+    u16 field_A;
+    u16 field_C;
+    u16 BLDCNT;
+    u16 BLDALPHA;
+    u16 BLDY;
+    s16 field_14;
+    s16 field_16;
+    s16 field_18;
+    s16 field_1A;
+    s16 field_1C;
```
Comment (Diegoisawesome) [inline src/battle_transition.c]: Should be BG0VOFS

File: src/battle_transition.c [hunk 3]
```diff
@@ -0,0 +1,4346 @@
...
+{
+    u8 i, r5;
+    u16* toStore;
+    s16 value;
+    s32 mergedValue;
+
+    sTransitionStructPtr->VBlank_DMA = FALSE;
+
+    toStore = gUnknown_02038C28[0];
+    r5 = task->tData1;
```
Comment (Diegoisawesome) [inline src/battle_transition.c]: Those fields should be named something related to BG0HOFS

Comment (Diegoisawesome) [inline src/battle_transition.c]: Does *= 2 work? If so, use it as it's more legible

Comment (Diegoisawesome) [inline src/battle_transition.c]: Does i & 1 work?

Comment (DizzyEggg) [reply src/battle_transition.c]: any suggestions?

Comment (Diegoisawesome) [inline src/battle_transition.c]: 2742: Were you unable to fix this?

Comment (Diegoisawesome) [reply src/battle_transition.c]: Maybe BG0HOFS_1 and _2? Nothing too complicated

Comment (DizzyEggg) [reply src/battle_transition.c]: huh, fix what? That was on gf's part :D

Comment (DizzyEggg) [reply src/battle_transition.c]: yup, changed it

Comment (DizzyEggg) [reply src/battle_transition.c]: nope, it must be lsl

Comment (Diegoisawesome) [reply src/battle_transition.c]: The compiler seriously turns *= 2 into a mul command? Usually it's smart enough to take multiplication by powers of two and make them lsl

Comment (DizzyEggg) [reply src/battle_transition.c]: it messes up registers

File: src/battle_transition.c [hunk 4]
```diff
@@ -0,0 +1,4346 @@
...
+    u16 BLDCNT;
+    u16 BLDALPHA;
+    u16 BLDY;
+    s16 field_14;
+    s16 field_16;
+    s16 field_18;
+    s16 field_1A;
+    s16 field_1C;
+    s16 field_1E; // unused
+    s16 field_20;
```
Comment (Diegoisawesome) [inline src/battle_transition.c]: Should probably be a bool16

File: src/battle_transition.c (hunk 2 above)
Comment (DizzyEggg) [reply src/battle_transition.c]: done

File: src/battle_transition.c (hunk 3 above)
Comment (DizzyEggg) [reply src/battle_transition.c]: done

**Pull request merged**